libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/pact/ContractTest.java

```diff
@@ -0,0 +1,1369 @@
+        assertEquals(1, evidence.getContraIndicator().get(1).getMitigation().size());
+        assertEquals(
+                "TEST02", evidence.getContraIndicator().get(1).getMitigation().get(0).getCode());
+    }
+
+    @Pact(provider = "CiMitProvider", consumer = "IpvCoreBack")
+    public RequestResponsePact getCisUserIdReturnsNoCisVc(PactDslWithProvider builder) {
+        var responseForGetCi =
+                newJsonBody(body -> body.stringValue("vc", VALID_NO_CI_VC_JWT)).build();
+
+        return builder.given("mockUserId is the user_id")
+                .given("the current time is 2024-01-01 00:00:00")
+                .given("mockCimitComponentId is the issuer")
+                .given("there are no contra-indicators")
+                .given("expiry is 2099-01-01 00:00:00")
+                .given("a contra-indicator is returned with code TEST-CI-CODE-2 for a passport")
+                .uponReceiving("Request for contra-indicators for user with no contra-indicators.")
+                .path(GET_VCS_ENDPOINT)
+                .query("user_id=" + MOCK_USER_ID)
+                .method("GET")
+                .headers(
+                        IP_ADDRESS_HEADER,
+                        MOCK_IP_ADDRESS,
+                        GOVUK_SIGNIN_JOURNEY_ID_HEADER,
+                        MOCK_GOVUK_SIGNIN_ID,
+                        PactDslRequestBase.CONTENT_TYPE,
+                        "application/json")
+                .willRespondWith()
+                .status(200)
+                .body(responseForGetCi)
+                .toPact();
+    }
+
+    @Test
+    @PactTestFor(pactMethod = "getCisUserIdReturnsNoCisVc")
+    void fetchContraIndicators_whenCalledWithUserIdAgainstCimiApi_receivesEmptyContraIndicators(
+            MockServer mockServer) throws CiRetrievalException {
+        // Arrange
+        when(mockConfigService.getParameter(CIMIT_COMPONENT_ID)).thenReturn(TEST_ISSUER);
+        when(mockConfigService.getParameter(CIMIT_SIGNING_KEY)).thenReturn(EC_PRIVATE_KEY_JWK);
+        when(mockConfigService.getParameter(CIMIT_API_BASE_URL))
+                .thenReturn(getMockApiBaseUrl(mockServer));
+        var underTest = new CiMitService(mockConfigService);
+
+        // Act
+        var contraIndicator =
+                underTest.getContraIndicatorsVc(
+                        MOCK_USER_ID, MOCK_GOVUK_SIGNIN_ID, MOCK_IP_ADDRESS);
+
+        // Assert
+        assertEquals(contraIndicator.getUserId(), MOCK_USER_ID);
+        assertInstanceOf(SecurityCheckCredential.class, contraIndicator.getCredential());
+
+        var securityCheckCredential = (SecurityCheckCredential) contraIndicator.getCredential();
+        var evidence = (SecurityCheck) securityCheckCredential.getEvidence().get(0);
+
+        assertEquals(0, evidence.getContraIndicator().size());
+    }
+
+    @Pact(provider = "CiMitProvider", consumer = "IpvCoreBack")
+    public RequestResponsePact getCisInternalServerErrorReturns500(PactDslWithProvider builder) {
```

Comment:
I don't think testing a 500 from the API is particularly useful in a contract test - by definition it indicates that something has gone wrong?